### ISSUE TITLE
Many-to-One SPS (Constellation)

### ISSUE THREAD

**Issue by darianvp** [MatthewGrim/Lunar_SPS#12]
Considering the question of how many solar power satellites are required in order to reduce the total blackout at a south pole target by an arbitrary amount (up to 100% of the original 52.5% of the time for which the target is eclipsed), where the single SPS can provide an approximate maximum reduction of 35% (with a 5000 km altitude or lower).

**Comment by darianvp** [comment]
The first attempt at establishing some information regarding the minimum number of satellites required for continuous coverage went as follows.

Using the SPS constrained design tool (Issue: #11), it was determined that an orbit with a 10 km perigee altitude and 4500 km apogee altitude is a high performing orbit design for a WPT link between the 100 kW transmitter option, and Sorato rover option. I chose this as the starting point for the initial constellation study. The active time for a single SPS in this orbit is 35.35% (17.15% remaining total blackout time)

I simulated the lighting and access of two different SPS constellations:
1. Dual SPS, same orbit, distributed 180 degrees apart in true anomaly
2. Triple SPS, same orbit, distributed 120 degrees apart in true anomaly

The total active time of the SPS constellations was determined by combining the individual access times of each satellite:

1. Total active time = 52.4297%. Remaining target blackout time = 0.0744%.
2. Total active time = 52.4323%. Remaining target blackout time = 0.0719%.

The remaining blackout events are discrete. One pair of events in particular persists for both constellation cases. It looks like (in STK) that this event is when the moon is in the eclipse of the Earth. Below is a screen cap of this scenario from a different SPS constellation design test, where the same blackout event persisted. Blue indicates portions of the orbit for which the satellite is not illuminated. 

Throughout the 2 year simulation (May 17 2018 - 2020) This event is repeated multiple times: 
Jul 27 2018, Jan 20 2019. Jul 16 2019, and Jan 10, 2020 (ie. twice per year). Generally the duration of the event is between 3 - 5 hours with the SPS constellations. Approximating the moons orbit as circular, the approximate duration of this event should be on the order of:

![codecogseqn 11](https://user-images.githubusercontent.com/38875772/41898293-f0591a0a-7929-11e8-9a3b-87b0e88dd883.gif)

In other words, it seems as though it might be impossible to fully eliminate blackout events at the lunar south pole with low altitude (< 5000 km) SPS orbiters. the best that can be done is a reduction to twice per year ~5 hour events (0.06% of the total time).

![eclipse_of_moon_jun26](https://user-images.githubusercontent.com/38875772/41895800-442b6950-7923-11e8-8897-ee32a52283f2.PNG)

**Comment by darianvp** [comment]
The second case which I tested is a high performing orbit design (450 km perigee 850 km apogee altitudes) for a 15kW transmitter targeting the same Sorato rover. I tested SPS constellations up to 6 satellites, distributed equally in true anomaly (360 / n where n is the number of satellites in the constellation).

1 SPS active time: 13.0382 %
1 SPS blackout time: 39.466 %

2 SPS active time: 26.0675 %
2 SPS blackout time: 26.4367 %

3 SPS active time: 39.102 %
3 SPS blackout time: 13.4022 %

4 SPS active time: 48.4292 %
4 SPS blackout time: 4.075 %

5 SPS active time: 51.7581 %
5 SPS blackout time: 0.7461 %

6 SPS active time: 52.4104 %
6 SPS blackout time: 0.0938 %

It appears like a hand-wavey answer to the question of how many satellites are required to "eliminate" (see previous comment) blackouts at the south pole target is:

![codecogseqn 13](https://user-images.githubusercontent.com/38875772/41899192-2153420a-792c-11e8-8c42-d08f34a99c79.gif)


Meaning if you divide the **original** blackout time at the target (52.5% for the south pole target) by the total active time (blackout reduction) provided by one SPS, you will find an approximate number of satellites required to reduce the total blackout time at the target significantly.

**Comment by darianvp** [comment]
I also wanted to investigate the assumption that the distribution of satellites as (360 / n) was justifiable in maximizing the constellation performance. The first look at this was calculating the total active of dual-sps constellations for the configuration described in the previous comment. I combine an SPS situated at 0deg true anomaly with satellites with the following initial true anomalies:

    true_anomalies = [60, 72, 90, 120, 144, 180, 216, 240, 270, 288, 300]

(generated for the study described in the previous comment) to form dual-sps constellations which had varying distance between the two satellites. Then I calculated the total active time of the constellations to assess the performance. The following plot was generated:

![compare_dual_sps_jun26](https://user-images.githubusercontent.com/38875772/41902202-291b9742-7933-11e8-8bb2-003c687870a8.png)

The "corners" of the plot occur at 120deg and 240deg.

**Comment by darianvp** [comment]
Based on the condition defined above the approximate number of SPS required to eliminate blackouts, the following plot was generated to determine the necessary constellation size for the SPS orbit design space studied in Issue #11:

![number_of_sps_to_eliminate_blackout_jun27](https://user-images.githubusercontent.com/38875772/41964140-92068660-79f9-11e8-9afb-1a9c891144e7.png)

In order to simulate all constellations of size between 2 and the number shown in the above plot for each design point in the orbital design space, an additional 6514 simulations are required. 

These simulations are needed to add the options of using an SPS constellation to further reduce the blackout at a target beyond the capabilities of a single SPS. 

The idea is that if a single SPS does not reduce the target blackout to the desired (constrained) level, then increase the number of SPS in the constellation size until the active times in the design space meet the constraint.

**Comment by darianvp** [comment]
The processed data for the polar and equatorial SPS orbital configurations are shown below. The south pole configuration leads to a "complete" (within 0.25%) irradication of the total eclipse (52.5%) at the south pole using two satellites within a significant portion of the orbit design space. The equatorial configuration, which has lower active periods in general, sees more or less an exact doubling of total active time with the introduction of the second SPS.

The maximum blackout duration plots have similar structure to the case with a single SPS. In the polar orbit, the magnitude of maximum blackout decreases (roughly halved), and in the equatorial case the extreme blackout duration at low perigee altitude are somewhat suppressed.

![2sps_southpolio](https://user-images.githubusercontent.com/38875772/43139089-264e6c9a-8f51-11e8-9fc4-aa4c54399ce9.png)

![2sps_equatoreo](https://user-images.githubusercontent.com/38875772/43139162-57c32824-8f51-11e8-9fdc-2bc1f526495f.png)
